app/web_framework/cluster_master_application.py

```diff
@@ -63,7 +63,7 @@ def __init__(self, cluster_master):
         api_v2 = [
             RouteNode(r'metrics', _MetricsHandler),
             RouteNode(r'version', _VersionHandler),
-            RouteNode(r'builds', _BuildsHandler).add_children([
+            RouteNode(r'builds', _BuildsPaginatedHandler).add_children([
```

Comment:
Let's call this something like `_v2BuildsHandler`. Just because pagination is the only difference now doesn't mean it will always be the only difference.